### PR TITLE
Call f.content.release() after updateWindow. Fixes #1728

### LINKED ISSUE #1728: linkerd 1.3.3 issue with gRPC
Issue Type:

- [x] Bug report
- [ ] Feature request

**What happened**:
I updated linkerd from `1.1.2` to `1.3.3` in DCOS 1.9 and started getting the following errors in gRPC clients:

from go client:
```
2017/12/05 17:30:05 rpc error: code = ResourceExhausted desc = grpc: received message larger than max (845559858 vs. 4194304)
```

from .net core client:
```
Unhandled Exception: Grpc.Core.RpcException: Status(StatusCode=Internal, Detail="Failed to deserialize response message.")
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Grpc.Core.Internal.AsyncCall`2.UnaryCall(TRequest msg)
   at Grpc.Core.DefaultCallInvoker.BlockingUnaryCall[TRequest,TResponse](Method`2 method, String host, CallOptions options, TRequest request)
   at Grpc.Core.Internal.InterceptingCallInvoker.BlockingUnaryCall[TRequest,TResponse](Method`2 method, String host, CallOptions options, TRequest request)
```

It looks like the message becomes invalid. The errors don't happen all the time, I need to call that endpoint multiple times. The message itself is not that big - around `40KB`.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
I tried running linkerd locally, but couldn't reproduce this issue.

**Anything else we need to know?**:

**Environment**:
- linkerd/namerd version, config files: linkerd 1.3.3, no namerd
- Platform, version, and config files (Kubernetes, DC/OS, etc): DCOS Enterprise 1.9
- Cloud provider or hardware configuration: AWS

linkerd configuration:
```
---
usage:
  enabled: false
admin:
  port: 9990
  ip: 0.0.0.0
namers:
- kind: io.l5d.marathon
  host: leader.mesos
  port: 443
  uriPrefix: "/marathon"
  prefix: "/io.l5d.marathon"
  tls:
    commonName: master.mesos
    trustCerts:
    - "/mnt/mesos/sandbox/.ssl/ca.crt"
routers:
- protocol: h2
  experimental: true
  identifier:
    kind: io.l5d.header.token
  dtab: "/ph=>/$/io.buoyant.rinet;/srv=>/$/io.buoyant.porthostPfx/ph;/svc=>/srv;/marathonId=>/#/io.l5d.marathon;/svc=>/$/io.buoyant.http.domainToPathPfx/marathonId"
  servers:
  - port: 4140
    ip: 0.0.0.0

```

### PR DESCRIPTION
**Pull request by vadimi**
This fixes gRPC deserialization errors described in #1728 that I'm having with linkerd `1.3.3` in DC/OS AWS environment.

### CODE REVIEW

**Comment by hawkw** [comment]
Hi @vadimi, it looks like your CI builds are failing due to an issue that was fixed on master in 9f037db949e5f61edf5837e94386db455ef732f0, would you be willing rebase your branch from the current master so we can get a clean CI build?

File: finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4Message.scala [hunk 1]
```diff
@@ -104,9 +104,12 @@ private[h2] object Netty4Message {
       val sz = f.content.readableBytes + f.padding
       val buf = ByteBufAsBuf(f.content.retain())
       val releaser: () => Future[Unit] = {
-        f.content.release()
-        if (sz > 0) () => updateWindow(sz)
-        else () => Future.Unit
+        () =>
+          {
+            val res = if (sz > 0) updateWindow(sz) else Future.Unit
```
Comment (hawkw) [inline finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4Message.scala]: Take it or leave it, but I think something like
```scala
      val releaser: () => Future[Unit] =
        () =>
          Future { f.content.release(); () }.before {
            if (sz > 0) updateWindow(sz) else Future.Unit
          }
```
might be slightly more idiomatic Scala.

**Comment by vadimi** [comment]
thank @hawkw, since it got merged into your local branch (not master) do you need me to do anything else with it?

**Comment by hawkw** [comment]
hi @vadimi, yeah, I merged your changes back into my branch so I could add some comments & test against the latest changes. I'm happy to take it from here, but if you'd like to get credit for the final merge commit, feel free to merge my changes back into your branch and open a PR against master?

**Comment by vadimi** [comment]
@hawkw, no, that's fine, I'm already pretty happy that the issue got finally fixed, it was driving me crazy. :)

**Comment by hawkw** [comment]
@vadimi cool, I've gone ahead and opened #1751; this should end up in the 1.3.4 release!

**Pull request merged**